### PR TITLE
Replace packr with golang's native embed package

### PR DESCRIPTION
**Pull request by kon-angelo**
**How to categorize this PR?**

/area control-plane
/kind cleanup
/os garden-linux

**What this PR does / why we need it**:
+ gardener revendoring at v1.21.0
+ Upgrades go version to v1.16
+ replaces packr2 with native `embed` package.

SInce the test framework from `g/g` depends on packr, the dependency cannot be completely eliminated for now, but it is only used on tests.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by kon-angelo** [comment]
It fails for now because `refs/meta/ci` is using golang v1.15.8 docker image.

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ############# builder
-FROM golang:1.15.8 AS builder
+FROM golang:1.16.2 AS builder
```
Comment (rfranzke) [inline Dockerfile:2]: Do you also want to update the alpine image?

File: pkg/generator/testfiles/embed.go [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package testfiles
```
Comment (rfranzke) [inline pkg/generator/testfiles/embed.go:1]: license header

**Review by rfranzke — COMMENTED** [review]
/invite @vpnachev

File: Dockerfile (hunk 1 above)
Comment (kon-angelo) [reply Dockerfile:2]: done

File: pkg/generator/testfiles/embed.go (hunk 2 above)
Comment (kon-angelo) [reply pkg/generator/testfiles/embed.go:15]: done

File: Dockerfile (hunk 1 above)
Comment (kon-angelo) [reply Dockerfile:2]: Actually should I change it to `eu.gcr.io/gardener-project/3rd/alpine` ?

Comment (rfranzke) [reply Dockerfile:2]: IIUC, this is no longer needed now that we use Kaniko

**Review by rfranzke — APPROVED** [review]
/lgtm

**Pull request merged**